### DISCUSSION TITLE
Accurate screen size info

### DISCUSSION THREAD

**Discussion by ghost** [plausible/analytics#1025] [Anything Else]
Google Analytics provide information on the screen sizes of visitors. Can Plausible provide more accurate screen sizes? Currently the generic screen size is given, but not specific screen sizes.

It would be even better if you could provide the size of the browser window (which obviously often isn't the same as the screen size). Is this possible?

**Comment by Vigasaurus** [comment]
For your first point, what would the UI look like to do something like that? Just a bar graph of every unique value?

For the second, that is how Plausible gathers widths; with `window.innerWidth` so it is browser width and not true monitor width all the time.

**Reply by ghost** [reply]
@Vigasaurus 

1.
Similar to Google Analytics, a list with the most visits at the top of the list:

800x600 = x visits
1440x900 = x visits
1920x1080 = x visits
2560x1440 = x visits
1366x768 = x visits
1280x720 = x visits
360x640 = x visits
1536x864 = x visits
375x667 = x visits
1280x800 = x visits

etc.

2.
Oh, but in the Plausible dashboard it says "Screen size" and uses icons of the different devices – which suggests Plausible is measuring screen sizes, rather than browser window sizes.

So if a visitor with a massive desktop screen, but a very small browser window, visits a site, it will show as a 'mobile' device, rather than a desktop device?

If this is the case, I think the text on the dashboard needs rewriting to make it clear the data is browser window size, not screen size.

**Reply by metmarkosaric** [reply]
> So if a visitor with a massive desktop screen, but a very small browser window, visits a site, it will show as a 'mobile' device, rather than a desktop device?

Yes, that's correct. You can read more here https://plausible.io/docs/devices

**Reply by ghost** [reply]
I therefore suggest the text in the Plausible dashboard is very misleading. It should be talking about 'browser window size', not 'device size' and it is misleading to show icons of devices.

What does this mean in the docs: "The actual width of the browser in pixels is discarded."?

**Reply by metmarkosaric** [reply]
There are some ideas on how we can improve that section in threads such as #379 and #157.

Like everything else we do with Plausible, any potentially sensitive data is discarded and never saved in our database or logs.

**Reply by ukutaht** [reply]
> I therefore suggest the text in the Plausible dashboard is very misleading. It should be talking about 'browser window size', not 'device size' and it is misleading to show icons of devices.

True, thanks for the suggestion.

> What does this mean in the docs: "The actual width of the browser in pixels is discarded."?

It means we don't store the actual measurements in the database. We calculate which category the screen falls into ('desktop', 'mobile' etc) and just store that category name as an enum.

**Reply by ghost** [reply]
> It means we don't store the actual measurements in the database. We calculate which category the screen falls into ('desktop', 'mobile' etc) and just store that category name as an enum.

Perhaps instead of saying "The actual width of the browser in pixels is discarded", you could say "we don't store the actual measurements in the database"? The later might be easier to understand?

**Reply by ghost** [reply]
> There are some ideas on how we can improve that section in threads such as #379.

I disagree thread 379 is still talking about screen sizes. Plausible is not measuring screen sizes, but browser window size, which is a totally different thing. A visitor with a 30 inch screen, but a browser window of only 200px width, will show up in the Plausible data as a 'mobile' user. Which is pretty weird.

**Reply by jeff-h** [reply]
I thought it was weird too until I considered what I might use such info for; if you're designing a website, you don't care what width someone *might* view it at, but rather what width someone *did* view it at. eg I have a 27" monitor and will not ever view a website fullscreen.

**Reply by ukutaht** [reply]
Exactly @jeff-h the reason I built it this way is to inform better development decisions. When designing a website, what matters is how people view the website. The total available screen real estate is irrelevant when it comes to css media queries.

**Reply by ghost** [reply]
> Exactly @jeff-h the reason I built it this way is to inform better development decisions. When designing a website, what matters is how people view the website. The total available screen real estate is irrelevant when it comes to css media queries.

I agree.

My point regards the **words** used in the dashboard. Don't call it 'screen size' and don't use the terms 'laptop' 'desktop' 'mobile' and 'tablet'.  Using these words communicates you are measuring screen sizes and devices, when you are not. You are measuring browser window size. A totally different thing.

Also, many of your customers will be used to using Google Analytics, which does measure screen size and devices. So if you use words such as 'devices' and 'screen size' 'mobile' 'tablet' etc they will think that you are measuring the same thing as Google Analytics.

**Comment by se-design-dk** [comment]
I think this is an excellent point.

Grouping all screen sizes/browser sizes above 1440px together in one category is a bit too broad.
Perhaps at least adding a 'Above 2560px' category would be really helpful.

**Comment by andrewholgate** [comment]
I agree that specific screen size data is very valuable and is a need addition to Plausible. The screen size data is used to determine the appropriate width breakpoints in a responsive UI design.

The only reason I still need GA installed on my projects is to capture this data.

**Comment by tpimh** [comment]
You can use a custom prop for this:
```js
window.onload = function() {
    plausible("Screen", {props: {Resolution: screen.width + "x" + screen.height}});
}
```

**Comment by DanielBielczyk** [comment]
Adding my support for accurate screen sizes. UI could be like detail on browser versions - just a list of unique resolutions.

**Comment by magoz** [comment]
It would be awesome to have support for more accurate screen sizes.

**Comment by coinbird-development** [comment]
We would also love to see this! It would be a good insight to see most used screen resolutions to optimize for.

**Reply by micah-redwood** [reply]
100% this. Knowing what your current user's viewport sizes are helps push designers to design for them and not just large screens or the default mobile viewport size that happens to be in their design tool at the time.

Would be nice to surface it in a way where you can clearly see what your most common viewport sizes are. I actually built a quick & dirty visualization tool at my last job using viewport size data exported from Google Analytics. You can even paste in your own data:
https://codepen.io/pranksinatra/pen/NWbqQjB

Would be awesome if Plausible supported UI plugins / extensions or something where you could write a component that displays Plausible data in the UI, or at least export it easily to a CSV.
